tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ServiceCommand.java

```diff
@@ -0,0 +1,294 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.core.modules;
+
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+import aQute.remote.api.Agent;
+import aQute.remote.api.Event;
+import aQute.remote.api.Supervisor;
+import aQute.remote.util.AgentSupervisor;
+
+/**
+ * @author Lovett Li
+ */
+public class ServiceCommand {
+
+    private String _serviceName;
+
+    public ServiceCommand()
+    {
+    }
+    public ServiceCommand( String serviceName )
+    {
+        _serviceName = serviceName;
+    }
+
+    public String[] execute() throws Exception
+    {
+        ServicesSupervisor supervisor = null;
+        String[] result = null;
+
+        synchronized( ServiceCommand.class )
```

Comment:
what does this do?  I don't think this makes sense and I don't know that this command needs any sychronization.  Don't add synchronization unless you know 100% for sure that it is required it only makes code more complex.